app/gui/src/controller/visualization.rs

```diff
@@ -155,7 +156,11 @@ impl Handle {
                 let js_code = self.language_server_rpc.read_file(path).await?.contents;
                 let wrap_error =
                     |err| Error::js_preparation_error(visualization.clone(), err).into();
-                visualization::java_script::Definition::new(project, &js_code)
+                let unknown_file_name = || String::from("unknown_file_name");
+                let file_name = path.file_name().cloned().unwrap_or_else(unknown_file_name);
```

Comment:
Actually, the error is rather unlikely here. I refactored it to log a warning to the console. We should not return an error if we can't extract the file name, because it will only affect source maps.